### DISCUSSION TITLE
Correct way to access Mongoose Models in TurboRepo?

### DISCUSSION THREAD

**Discussion by MattHeslington** [vercel/turborepo#2723] [Help] [answered]
Hello,

I have three apps|repos which all access the same MongoDB instance through Mongoose. At the moment I have around 20 large data Model files, and each time a model is updated in one repo I have to manually update the model file in my other apps too. It's ridiculous.  

I would like to move the three independent repos to a turborepo build where each 'app' has access to a single set of Mongoose data models, eliminating the duplication. I wonder if anyone has any experience of using a single set of Mongoose models in a TurboRepo? 

As I see it, each 'app' will have its own install of Mongoose, I just need a central Models folder that each app can import from e.g.

```
import Users from '~models/User'
```

I have tried creating a Models folder in the packages directory and then linking to it in my app's package.json but this just broke any pnpm installs or removals, giving the error 'The "path" argument must be of type string'

```
"dependencies": {
       ...
        "ui": "*",
        "models": "*"
    },
```

Any help would be very much appreciated. This is my first monorepo build so I do apologise if it's really simple etc..

Cheers,
Matt

**Comment by anthonyshew** [comment, marked as answer]
pnpm expects that you use `workspace:*` rather than `*`.
